backend/src/hatchling/metadata/spec.py

```diff
@@ -174,7 +174,7 @@ def project_metadata_from_core_metadata(core_metadata: str) -> dict[str, Any]:
                     left, _, right = marker
                     if left.value == 'extra':
                         extra = right.value
-                        del markers[i]
+                        del markers[i]  # noqa: B909
```

Comment:
This is safe _currently_ because there is always a `break` after the deletion, but it would be preferable to do this safely rather than suppress the warning?

Reply:
Let me take a quick crack at it...

Reply:
I will PR your branch soon later today with a rewrite, I have something working but I have some meetings.